src/Shouldly/MessageGenerators/DictionaryShouldContainKeyAndValueMessageGenerator.cs

```diff
@@ -14,6 +15,9 @@ public override bool CanProcess(IShouldlyAssertionContext context)
 
         public override string GenerateErrorMessage(IShouldlyAssertionContext context)
         {
+            Debug.Assert(context.Actual is IDictionary);
+            Debug.Assert(context.Key is object);
```

Comment:
⚠ Not sure how you'd like to handle these. This method is relying on the fragile assumption that remote code will have set these fields, and these asserts silence the warnings this causes now that NRTs are enabled. The way I would handle it would be to require these values to be passed to the DictionaryShouldContainKeyAndValueMessageGenerator constructor.